lua/pac3/editor/client/parts.lua

```diff
@@ -330,7 +331,7 @@ do -- menu
 				pnl:SetImage(groupData.icon)
 			end
 
-			local trap = false
```

Comment:
Why'd this local get moved?

Reply:
https://github.com/CapsAdmin/pac3/blob/0fe2506c40ff2e907adffeba230bbad0ae08b1f6/lua/pac3/editor/client/parts.lua#L273

In the original code it was being defined above it, thus creating a global

Reply:
Oh ok, I see it now. So the code has been completely broken I think. Probably need to test whatever that trap var is used for.

Reply:
The rest of the changes look fine.

Reply:
Now that im looking at it, i don't think this code ever runs? `pace.AddRegisteredPartsToMenu` is called by 2 functions `pace.OnPartMenu` and `pace.OnNewPartMenu` don't seem to be called anywhere? what the hell

Reply:
`pace.OnCreatePart` `pace.OnPartSelected` `pace.OnVariableChanged` `pace.OnPartMenu` `pace.OnNewPartMenu` `pace.OnHoverPart` don't seem to be used? i must be missing something

Reply:
> `pace.OnCreatePart` `pace.OnPartSelected` `pace.OnVariableChanged` `pace.OnPartMenu` `pace.OnNewPartMenu` `pace.OnHoverPart` don't seem to be used? i must be missing something

they are used "magically" by pace.RunHook("PartSelected") or something like that

Reply:
Probably string concatenation being used to call them `"On"..hookname`, which is dumb and needs to be removed someday if so.

Reply:
![image](https://user-images.githubusercontent.com/69946827/201435022-5ac093b5-9cab-41f6-b6cd-a1f675ce9347.png)
yup

Reply:
The `trap` move seems to work fine